packages/core/src/js/sdk.tsx

```diff
@@ -66,13 +66,13 @@ export function init(passedOptions: ReactNativeOptions): void {
     if (!dsn) {
       return undefined;
     }
-    try {
-      const url = new URL(dsn);
-      return `${url.protocol}//${url.host}`;
-    } catch (e) {
-      logger.error('Failed to extract url from DSN', e);
+    const dsnComponents = makeDsn(dsn);
+    if (!dsnComponents) {
+      logger.error('Failed to extract url from DSN: ', dsn);
       return undefined;
     }
+    const port = dsnComponents.port ? `:${dsnComponents.port}` : '';
```

Comment:
This DSN return is used on the following code logic:
```typescript
(dsn && url.startsWith(dsn)))) {
      return null;
    }
```

Q: Isn't the port here redundant?  Since we check the initial DSN path, it should filter if we set nor not the port, making the port parameter redundant.

I: Consider removing the port parameter, unless it's really required.